File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/LogoutCommand.java
package ro.razvan.java.dagger.atm;

import javax.inject.Inject;

public class LogoutCommand implements NoArgsCommand {

    public static final String COMMAND_KEY_LOGOUT = "logout";

    private final Outputter outputter;

    @Inject
    public LogoutCommand(Outputter outputter) {
        this.outputter = outputter;
    }

    @Override
    public Result handle() {
        outputter.output("Logging out...");
        return Result.inputCompleted();
    }

    @Override
    public String key() {
        return COMMAND_KEY_LOGOUT;
    }
}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/LogoutCommand.kt
package ro.razvan.kotlin.dagger.atm

import ro.razvan.kotlin.dagger.atm.Command.Result
import javax.inject.Inject

const val COMMAND_KEY_LOGOUT = "logout"

class LogoutCommand @Inject constructor(
    private val outputter: Outputter
) : NoArgsCommand {

    override fun handle(): Result {
        outputter("Logging out...")
        return Result.inputCompleted()
    }

    override val key: String
        get() = COMMAND_KEY_LOGOUT

}